ui/app/components/auth/form-template.js

```diff
@@ -0,0 +1,186 @@
+
+  // auth login variables
+  @tracked selectedAuthMethod = 'token';
+  @tracked errorMessage = null;
+
+  displayName = (type) => {
+    const displayName = ALL_LOGIN_METHODS?.find((t) => t.type === type)?.displayName;
+    return displayName || type;
+  };
+
+  constructor() {
+    super(...arguments);
+    this.fetchMounts.perform();
+  }
+
+  get availableMethodTypes() {
+    return supportedTypes(this.version.isEnterprise);
+  }
+
+  get formComponent() {
+    const { selectedAuthMethod } = this;
+    // isSupported means there is a component file defined for that auth type
+    const isSupported = this.availableMethodTypes.includes(selectedAuthMethod);
+    const formFile = () => (['oidc', 'jwt'].includes(selectedAuthMethod) ? 'oidc-jwt' : selectedAuthMethod);
+    const component = isSupported ? formFile() : 'base';
+
+    // an Auth::Form::<Type> component exists for each method in supported-login-methods
+    return `auth/form/${component}`;
+  }
+
+  get namespaceInput() {
+    const namespaceQP = this.args.namespace;
+    if (this.flags.hvdManagedNamespaceRoot) {
+      // When managed, the user isn't allowed to edit the prefix `admin/` for their nested namespace
+      const split = namespaceQP.split('/');
+      if (split.length > 1) {
+        split.shift();
+        return `/${split.join('/')}`;
+      }
+      return '';
+    }
+    return namespaceQP;
+  }
+
+  get renderTabs() {
+    // renders tabs if listing visibility is set (auth tabs exist)
+    // and user has NOT clicked "Sign in with other"
+    if (this.authTabs && !this.showOtherMethods) {
+      return true;
+    }
+    return false;
+  }
+
+  @action
+  handleAuthSelect(element, event, idx) {
+    if (element === 'tab') {
+      this.setAuthTypeFromTab(idx);
+    } else {
+      this.selectedAuthMethod = event.target.value;
+    }
+  }
+
+  setAuthTypeFromTab(idx) {
+    this.selectedAuthMethod = Object.keys(this.authTabs)[idx];
+    this.selectedTabIndex = idx;
+  }
+
+  @action
+  handleError(message) {
+    this.errorMessage = message;
+  }
+
+  @action
+  toggleView() {
+    this.showOtherMethods = !this.showOtherMethods;
+
+    if (this.renderTabs) {
+      // reset selected auth method to first tab
+      this.handleAuthSelect('tab', null, 0);
+    } else {
+      // all methods render, reset dropdown
+      this.selectedAuthMethod = 'token';
+    }
+  }
+
+  @action
+  async handleNamespaceUpdate(event) {
+    // update query param
+    this.args.handleNamespaceUpdate(event.target.value);
+    // reset tabs
+    this.authTabs = null;
+    // fetch mounts for that namespace
+    this.fetchMounts.perform(500);
+  }
+
+  fetchMounts = restartableTask(
+    waitFor(async (wait) => {
+      // task is `restartable` so if the user starts typing again,
+      // it will cancel and restart from the beginning.
+      if (wait) await timeout(wait);
+
+      try {
+        // clear ember data store before re-requesting.. :(
+        this.store.unloadAll('auth-method');
```

Comment:
🫠 this fetch logic is essentially the same as here: https://github.com/hashicorp/vault/blob/cd0cd0a175ab6bc762df7bd5c2cc9f40b26d9402/ui/app/components/auth-form.js#L199-L227

I avoided refactoring further because the removal of ember-data this logic will drastically improve and simplify. Additionally, I'm making changes as incrementally as possible to avoid regressions.